### PR TITLE
Reduce the nproc when building FA on 8.9

### PR DESCRIPTION
**Pull request by huydhn**
The newly introduce sm89 build is failing consistently in trunk now because of OOM https://github.com/pytorch/pytorch/actions/runs/12186328178/job/33994606556.  I suspect that FlashAttention is the cause.

### Testing

CI to see if the build works.

### CODE REVIEW

**Review by malfet — APPROVED** [review]
If it works, that SGTM, but something change in our infra to cause some many OOMs recently...

File: .ci/pytorch/build.sh [hunk 1]
```diff
@@ -191,7 +191,7 @@ fi
 
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" == *"8.6"* || "$TORCH_CUDA_ARCH_LIST" == *"8.0"* ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" > "8.0" || "$TORCH_CUDA_ARCH_LIST" == "8.0" ]]; then
```
Comment (malfet) [inline .ci/pytorch/build.sh:194]:
```suggestion
if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```

File: .ci/pytorch/build.sh [hunk 2]
```diff
@@ -191,7 +191,7 @@ fi
 
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" > "8.0" || "$TORCH_CUDA_ARCH_LIST" == "8.0" ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```
Comment (huydhn) [inline .ci/pytorch/build.sh:194]: Somehow, `>=` is a wrong syntax when I test it out locally, thus the weird use of `>` and `==`.  Let's see

Comment (huydhn) [reply .ci/pytorch/build.sh:194]: Yeah, bash doesn't like this syntax and shows `syntax error in conditional expression`.  Let me fix this

File: .ci/pytorch/build.sh [hunk 3]
```diff
@@ -191,7 +191,7 @@ fi
 
 # We only build FlashAttention files for CUDA 8.0+, and they require large amounts of
 # memory to build and will OOM
-if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" == *"8.6"* || "$TORCH_CUDA_ARCH_LIST" == *"8.0"* ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" >= "8.0" ]]; then
```
Comment (malfet) [inline .ci/pytorch/build.sh:194]:
```suggestion
if [[ "$BUILD_ENVIRONMENT" == *cuda* ]] && [[ "$TORCH_CUDA_ARCH_LIST" -ge "8.0" ]]; then
```

File: .ci/pytorch/build.sh (hunk 2 above)
Comment (malfet) [reply .ci/pytorch/build.sh]: Let's see if `-ge` will do the trick

**Pull request closed without merging**